lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -96,7 +96,11 @@ VarDecl *tf::getFieldIfContainsSingleField(NominalTypeDecl *decl) {
 }
 
 bool tf::isTensorHandle(SILType ty) {
-  return (bool)isTensorHandle(ty.getASTType());
+  return isTensorHandle(ty.getASTType());
+}
+
+bool tf::isOpaqueHandle(SILType ty) {
+  return isOpaqueHandle(ty.getASTType());
```

Comment:
do we expect `isOpaqueHandle()` from TensorFlow.h to have other call sites? Consider folding that func into the func here for now.

Reply:
Yes, we do. The type checker will need to be extended to use `isOpaqueHandle` to diagnose **host** functions taking opaque handles or returning opaque handles as they are not allowed by definition.

Reply:
Ah I see doing that will be a future PR. SGTM.

Reply:
Filed [SR-8503](https://bugs.swift.org/browse/SR-8503) to future work.